### ISSUE TITLE
question: missing server connection error to cluster

### ISSUE THREAD

**Issue by dtzar** [k8sgpt-ai/k8sgpt#225]
Checklist:

* [x] I've searched for similar issues and couldn't find anything matching
* [x] I've included steps to reproduce the bug.
* [x] I've included the version of Kubernetes and k8sgpt.

### Subject of the issue

k8sgpt can't function, despite a working connection to the cluster with kubectl.  

### Your environment

* Version of Kubernetes
Client Version: v1.26.1
Kustomize Version: v4.5.7
Server Version: v1.25.5

* Host OS and its version / If windows, is it WSL?
WSL, Ubuntu 22.04 LTS

* Version of k8sgpt
0.2.0

### Steps to reproduce
Not sure how to reproduce on your side.  I have no problems doing any operation from WSL against the cluster using the current context I have (admin) - even installing helm charts, etc.

### Expected behaviour
Connect to the kubernetes cluster or give more information.

### Actual behaviour

Any command (i.e. `generate` or `analyze`) results in this message:
`Error initialising kubernetes client: invalid configuration: [context was not found for specified context: myclustername, cluster has no server defined]`

**Comment by AlexsJones** [comment]
Just to clarify, you have problems installing helm charts? So outside the context of `k8sgpt` also?

**Comment by dtzar** [comment]
No - any k8sgpt command does not work and gives this error.
Is there a way to get debug information or more specific details on what it's looking for in the kubeconfig other than "server defined"?

**Comment by thomasstxyz** [comment]
Let me help with troubleshooting.

My working `~/.kube/config` looks like this:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tL...
    server: https://127.0.0.1:63575
  name: kind-dev
contexts:
- context:
    cluster: kind-dev
    user: kind-dev
  name: kind-dev
current-context: kind-dev
kind: Config
preferences: {}
users:
- name: kind-dev
  user:
    client-certificate-data: LS0tL...
    client-key-data: LS0tL...
```

If I would delete the context item:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tL...
    server: https://127.0.0.1:63575
  name: kind-dev
contexts:
current-context: kind-dev
kind: Config
preferences: {}
users:
- name: kind-dev
  user:
    client-certificate-data: LS0tL...
    client-key-data: LS0tL...
```

then I receive exactly the same error message:

```
Error initialising kubernetes client: invalid configuration: [context was not found for specified context: kind-dev, cluster has no server defined]
```

How does your kubeconfig look like?

**Comment by dtzar** [comment]
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: DATA+OMITTED
    server: https://ascap####:443
  name: ascapz
contexts:
- context:
    cluster: ascapz
    namespace: dev
    user: clusterUser_ascapz_ascapz
  name: ascapz
current-context: ascapz
kind: Config
preferences: {}
users:
- name: clusterUser_ascapz_ascapz
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - get-token
      - --environment
      - AzurePublicCloud
      - --server-id
      - 6dae####
      - --client-id
      - 80faf####
      - --tenant-id
      - a29d3####
      - --login
      - devicecode
      command: kubelogin
      env: null
      interactiveMode: IfAvailable
      provideClusterInfo: false
```

**Comment by AlexsJones** [comment]
Sorry I haven't responded, Kubecon this week meant everything is a bit behind

**Comment by dtzar** [comment]
No worries - if there is more information I can provide (i.e. debug, etc.) let me know.

**Comment by AlexsJones** [comment]
I did a bit of testing on this and in one scenario was able to reproduce it by tearing down and standing up a cluster then changing terminal.


The fix for me was to `export KUBE_CONFIG_PATH=/path/to/.kube/config` then `k8sgpt analyze` in the same shell. 
Let me know if that changes the output at all!

**Comment by dtzar** [comment]
Your steps didn't work, same result.
In my `.bashrc` I have `export KUBECONFIG=/different/path/config` 

I decided to rename the default `~/.kube/config` file and 
I discovered it appears you have a hard-coded check for the config file.  I then got the result:
`Error initialising kubernetes client: stat ~/.kube/config: no such file or directory`

Yet I still have access to do kubectl commands against my cluster.
